snapd-glib/snapd-notices.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2 or version 3 of the License.
+ * See http://www.gnu.org/copyleft/lgpl.html the full text of the license.
+ */
+
+#pragma once
+
+#if !defined(__SNAPD_GLIB_INSIDE__) && !defined(SNAPD_COMPILATION)
+#error "Only <snapd-glib/snapd-glib.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include "snapd-notice.h"
+
+G_BEGIN_DECLS
+
+#define SNAPD_TYPE_NOTICES  (snapd_notices_get_type ())
+
+G_DECLARE_FINAL_TYPE (SnapdNotices, snapd_notices, SNAPD, NOTICES, GObject)
+
+SnapdNotices      *snapd_notices_new           (GPtrArray *data);
+
+guint64            snapd_notices_get_n_notices (SnapdNotices *notices);
+
+SnapdNotice       *snapd_notices_get_notice    (SnapdNotices *notices,
+                                                guint64       notice);
```

Comment:
But the Qt API was already like this, so, in fact, it is more consistent to use this.

About the guint64->gsize... that value is the notice number, not the size... I'll change the name.

Reply:
The Qt API is like this because this seemed the simplest way to map the GObjects to Qt. My point is we should make any new API match the existing API for consistency, and then change it all in one release if we think the old method is unsuitable.

gsize is confusing, but it's the safest type to use for an index - it's always big enough to index everything in a given architecture. In comparison, guint64 is overly specific - we don't need 64 bits of precision exactly. See https://www.geeksforgeeks.org/size_t-data-type-c-language/

Reply:
Oh, so you meant consistency with the rest of the snapd-glib API! Ok, that makes sense. I'll revert that code.

And about gsize, you are right again. I'll change that too.

Reply:
(the gsize one I changed it to uint, because in GPtrArray the length field is a guint).